src/decode.h

```diff
@@ -250,7 +276,68 @@ static inline uint32_t sign_extend_b(const uint32_t x)
     return (int32_t) ((int8_t) x);
 }
 
+/* hash function is used when mapping address into the block map */
+static inline uint32_t hash(size_t k)
```

Comment:
Ditto. The hash function is implementation-specific. Don't put it in the header unless it is common.